presto-main/src/main/java/com/facebook/presto/operator/annotations/AnnotationHelpers.java

```diff
@@ -98,4 +100,12 @@ else if (comparableRequired.contains(name)) {
         }
         return typeVariableConstraints.build();
     }
+
+    public static void validateSignaturesCompatibility(Optional<Signature> signatureOld, Signature signatureNew)
```

Comment:
The first argument is guarantee to be present (in the two callsites to this function). Remove `Optional` from the signature?